File: packages/php/common/src/Helpers/VaultUtils.php
<?php

namespace Common\Helpers;

class VaultUtils
{
    private static $envToWebVaultUrlMap = [
        Environment::LOCAL => 'http://localhost:3001',
        Environment::DEVELOPMENT => 'https://vault.dev.affinidi.com',
        Environment::PRODUCTION => 'https://vault.affinidi.com',
    ];

    private static $sharePath = '/login';
    private static $claimPath = '/claim';

    public static function buildShareLink(string $request, string $clientId): string
    {
        $env = EnvironmentUtils::fetchEnvironment();
        $baseUrl = self::$envToWebVaultUrlMap[$env] ?? self::$envToWebVaultUrlMap[Environment::PRODUCTION];

        $queryString = http_build_query([
            'request' => $request,
            'client_id' => $clientId,
        ]);

        return "$baseUrl" . self::$sharePath . "?$queryString";
    }

    public static function buildClaimLink(string $credentialOfferUri): string
    {
        $env = EnvironmentUtils::fetchEnvironment();
        $baseUrl = self::$envToWebVaultUrlMap[$env] ?? self::$envToWebVaultUrlMap[Environment::PRODUCTION];

        $queryString = http_build_query([
            'credential_offer_uri' => $credentialOfferUri,
        ]);

        return "$baseUrl" . self::$claimPath . "?$queryString";
    }
}

File: packages/dart/affinidi_common/lib/src/vault_utils.dart
import 'dart:core';

import 'package:affinidi_common/affinidi_common.dart';

const envToWebVaultUrlMap = {
  Environment.local: 'http://localhost:3001',
  Environment.development: 'https://vault.dev.affinidi.com',
  Environment.production: 'https://vault.affinidi.com',
};

const String sharePath = '/login';
const String claimPath = '/claim';

class VaultUtils {
  static String fetchWebVaultUrl([Environment? env]) {
    env ??= EnvironmentUtils.fetchEnvironment();
    return envToWebVaultUrlMap[env]!;
  }

  static String buildShareLink(String request, String clientId,
      [Environment? env]) {
    final vaultUrl = fetchWebVaultUrl(env);
    Map<String, String> params = {
      'request': request,
      'client_id': clientId,
    };
    String queryString = Uri(queryParameters: params).query;
    return '$vaultUrl$sharePath?$queryString';
  }

  static String buildClaimLink(String credentialOfferUri, [Environment? env]) {
    final vaultUrl = fetchWebVaultUrl(env);
    Map<String, String> params = {
      'credential_offer_uri': credentialOfferUri,
    };
    String queryString = Uri(queryParameters: params).query;
    return '$vaultUrl$claimPath?$queryString';
  }
}
